src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -456,6 +456,10 @@ private static async Task<object> ExecuteHubMethod(ObjectMethodExecutor methodEx
                 {
                     await (Task)methodExecutor.Execute(hub, arguments);
                 }
+                else if (methodExecutor.MethodReturnType == typeof(ValueTask))
```

Comment:
@anurse I can remove this if it makes it easier to get through ship-room (as it will be test only without this)

Reply:
If everything works without it, yank it. If we want to avoid the boxing, go ahead and put that in `master`.